### DISCUSSION TITLE
realtime with RLS not working? or something missing here?

### DISCUSSION THREAD

**Discussion by malalecherocks** [supabase/supabase#8129] [Questions] [answered]
Hi,

I have activated realtime database to the table **message** and it works perfectly, but the moment I activate RLS to the table, it stop working and the policy is all true:

ALL
all true
true

and my subsription:
```
const mySubscription = supabase
		.from('message')
		.on('*', (payload) => {
			console.log('Change received!', payload);
			getMessages();
		})
		.subscribe();
```

Am I missing something here? If I do a refresh it loads the messages with getMessages() which means RLS is working well. @GaryAustin1 🙏

**Comment by GaryAustin1** [comment, marked as answer]
If your gutsy you might want to try the fix in this thread for realtime not working with RLS on some instances...
https://github.com/supabase/realtime/issues/265#issuecomment-1196604286
If not, and you are using current client sdk's I would contact support.  Your RLS settings should work for anon (non signed in user) which is the other common reasons for not working with RLS.

**Reply by malalecherocks** [reply]
Is there any question that you don’t know the answer? Thanks! It worked out well!

**Reply by w3b6x9** [reply]
@GaryAustin1 knows all!
